ui/src/components/LoginForm/SessionTable.jsx

```diff
@@ -5,6 +5,35 @@ import { LuExternalLink } from 'react-icons/lu';
 import SessionDateTime from './SessionDateTime';
 import styles from './SessionTable.module.css';
 
+function isDateInRange(actualStartDate, actualEndDate, rangeInDays) {
+  // Helper function to parse YYYYMMDD format into Date object
+  function parseDate(yyyymmdd) {
+    const year = yyyymmdd.slice(0, 4);
+    const month = yyyymmdd.slice(4, 6);
+    const day = yyyymmdd.slice(6, 8);
+    return new Date(`${year}-${month}-${day}`);
+  }
+
+  const startDate = parseDate(actualStartDate);
+  const endDate = parseDate(actualEndDate);
+
+  // Get today's date without time
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  // Calculate the lower and upper bounds based on the rangeInDays
+  const lowerBound = new Date(today);
+  lowerBound.setDate(today.getDate() - rangeInDays);
```

Comment:
`getDate()` returns the day number within the month, so it could lead to a session from the month before/after to be highlighted. Even if unlikely, maybe a simpler way would be to perform all the calculations with timestamps rather than date objects. For instance, assuming you include the time:

```js
const oneDay = 24 * 60 * 1000
const now = Date.now();

return startDate.getTime() <= now + oneDay && endDate.getTime() >= now - oneDay;
```

Reply:
I understood that JavaScript handles month overflow correctly when using setDate()

```
const today = new Date();
today.setHours(0, 0, 0, 0);
> 1740438000000

const lowerBound = new Date(today);
rangeInDays = 7

lowerBound.setDate(today.getDate() + rangeInDays);
> 1741042800000

lowerBound
> Date Tue Mar 04 2025 00:00:00 GMT+0100 (Central European Standard Time)

today
> Date Tue Feb 25 2025 00:00:00 GMT+0100 (Central European Standard Time)
```

Reply:
Riiight, okay, thanks! Phew, date manipulation always hurts my brain.